Lib/site.py

```diff
@@ -458,13 +458,12 @@ def venv(known_paths):
     env = os.environ
     if sys.platform == 'darwin' and '__PYVENV_LAUNCHER__' in env:
         executable = sys._base_executable = os.environ['__PYVENV_LAUNCHER__']
-    elif sys.platform == 'win32' and '__PYVENV_LAUNCHER__' in env:
+    elif sys.platform == 'win32':
         executable = sys.executable
-        import _winapi
-        sys._base_executable = _winapi.GetModuleFileName(0)
-        # bpo-35873: Clear the environment variable to avoid it being
-        # inherited by child processes.
-        del os.environ['__PYVENV_LAUNCHER__']
+        if '__PYVENV_LAUNCHER__' in env:
+            # bpo-35873: Clear the environment variable to avoid it being
+            # inherited by child processes.
+            del os.environ['__PYVENV_LAUNCHER__']
```

Comment:
Going to move this bit into `getpathp.c`, which is probably where it should have been the whole time.